### PR TITLE
recast loralayer, norm, lmhead + embed token weights per original qlora

### PR DESCRIPTION
**Pull request by winglian**
see https://github.com/artidoro/qlora/blob/main/qlora.py#L396-L405

### CODE REVIEW

**Comment by NanoCode012** [comment]
Should this be for qlora only or lora too and difference from the one near kbit?

**Comment by winglian** [comment]
I think both, since lora and lora only differ really in having a quantized frozen base model.

File: setup.py [hunk 1]
```diff
@@ -29,5 +29,8 @@
             "flash-attn",
             "deepspeed",
         ],
+        "peft": [
+            "peft @ git+https://github.com/huggingface/peft.git",
+        ],
```
Comment (winglian) [inline setup.py:34]: we do it this way b/c we don't install Peft with setup.py as the gptq variant has a different Peft version pinned.

Comment (NanoCode012) [reply setup.py:34]: Should we just add the pip install line to the `test.yml` instead of adding it here?

Comment (winglian) [reply setup.py:34]: I really want to deprecate GPTQ support 😭 , then this wouldn't be an issue

Comment (NanoCode012) [reply setup.py:34]: I'm wondering, how many people actually do use it?

Comment (mhenrichsen) [reply setup.py:34]: Time to do another poll on github maybe?

**Review by tmm1 — APPROVED** [review]

**Pull request merged**